content/class/Media.md

```diff
@@ -0,0 +1,213 @@
+
+```c#
+public string alt_text { get; }
+```
+
+A description of the media for display on hover.
+
+| API   | Twitter 1              | Twitter 2 | Mastodon |
+| ----- | ---------------------- | --------- | -------- |
+| Field | `include_ext_alt_text` |           | `url`    |
+| Type  | string                 |           | string   |
+
+### `height` {#property_height}
+
+![API: Twitter 1](https://img.shields.io/badge/API-Twitter%201-lightgrey?style=flat-square) ![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public int height { get; }
+```
+
+The height of the media.
+
+| API   | Twitter 1       | Twitter 2 | Mastodon               |
+| ----- | --------------- | --------- | ---------------------- |
+| Field | `sizes/large/h` | `height`  | `meta/original/height` |
+| Type  | int             | int       | int                    |
+
+### `width` {#property_width}
+
+![API: Twitter 1](https://img.shields.io/badge/API-Twitter%201-lightgrey?style=flat-square) ![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public int width { get; }
+```
+
+The width of the media.
+
+| API   | Twitter 1       | Twitter 2 | Mastodon              |
+| ----- | --------------- | --------- | --------------------- |
+| Field | `sizes/large/w` | `width`   | `meta/original/width` |
+| Type  | int             | int       | int                   |
+
+### `load_preview_progress` {#property_load_preview_progress}
+
+![API: Internal](https://img.shields.io/badge/API-Internal-green?style=flat-square)\
+
+```c#
+public double progress_load_preview { get; }
+```
+
+How far the loading of the preview image is completed.
+
+### `load_media_progress` {#property_load_media_progress}
+
+![API: Internal](https://img.shields.io/badge/API-Internal-green?style=flat-square)\
+
+```c#
+public double progress_load_preview { get; }
+```
+
+How far the loading of the media is completed.
+
+## Methods
+
+### `load_preview` {#method_load_preview}
+
+![API: Internal](https://img.shields.io/badge/API-Internal-green?style=flat-square)\
+
+```c#
+public async void load_preview ()
+```
+
+Loads the preview image. Activates [`load_preview_completed`](#signal_load_preview_completed) on completion.
+
+### `load_media` {#method_load_media}
+
+![API: Internal](https://img.shields.io/badge/API-Internal-green?style=flat-square)\
+
+```c#
+public abstract async void load_media ()
+```
+
+Loads the media and returns it. Activates [`load_media_completed`](#signal_load_media_completed) on completion.
+
+> Neither `Media` nor it's subclass stores the full media. Instead the function loads it and hands it over to the Media Viewer, which clears the media from memory after it is closed.
+
+## Signals
```

Comment:
I think we need progress signals as well, so that we can update load progress percentages etc